File: 프로그래머스/lv1/12901. 2016년/2016년.py
def solution(a, b):
    
    days = ['FRI','SAT','SUN','MON','TUE','WED','THU']
    months = [31, 29, 31, 30, 31, 30,31, 31, 30, 31, 30, 31]
    
    total_days = 0
    for i in range(a-1):
        total_days += months[i]
    
    total_days += b
    
    return days[total_days % 7 - 1]
    

File: 프로그래머스/1/12901. 2016년/2016년.java
import java.util.*;
class Solution {
    public String solution(int a, int b) {
        String answer = "";
        
        HashMap<Integer, Integer> map = new HashMap<>() {{
            put(1, 31);
            put(2, 29);
            put(3, 31);
            put(4, 30);
            put(5, 31);
            put(6, 30);
            put(7, 31);
            put(8, 31);
            put(9, 30);
            put(10, 31);
            put(11, 30);
            put(12, 31);
        }};
        
        int day = 0;
        for(int i = 1; i < a; i++) {
            day += map.get(i);
        }
        
        day += b;
        //System.out.println(day + ", " + day % 7);
        HashMap<Integer, String> days = new HashMap<>() {{
            put(0, "THU");
            put(1, "FRI");
            put(2, "SAT");
            put(3, "SUN");
            put(4, "MON");
            put(5, "TUE");
            put(6, "WED");
        }};
        // 금, 토, 일, 월, 화, 수, 목
        return days.get(day % 7);
    }
}